In [1]:
# Load model
import joblib
model = joblib.load('/content/DBS_SGD_Prediction.pkl')

In [2]:
model.coef_

array([[-50.60094302]])

- This is the slope (weight) learned by the model

- Since you used one feature (SGD), there is one coefficient

**Interpretation**:

> For every 1 unit increase in SGD, the predicted DBS value decreases by ~50.6 units

In [3]:
model.intercept_

array([90.22858515])

- This is the bias term

- Value of DBS when SGD = 0 (theoretical baseline)

The full prediction equation

The model represents:

`DBS = (−50.6009) × SGD + 90.2286`


In [9]:
prediction = model.predict([[1.29]])
prediction[0][0]

/usr/local/lib/python3.12/dist-packages/sklearn/utils/validation.py:2739: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


np.float64(24.953368654305933)

- Calls the trained LinearRegression model

- Uses 1.29 as the input value for the feature (SGD)

- Returns the predicted DBS value

This is using the model equation:

`DBS=(−50.6009)×1.29+90.2286≈24.95`